src/Mod/Assembly/App/AssemblyObject.cpp

```diff
@@ -783,34 +790,54 @@ std::vector<App::DocumentObject*> AssemblyObject::getGroundedParts()
 
         if (propObj) {
             App::DocumentObject* objToGround = propObj->getValue();
-            groundedObjs.push_back(objToGround);
+            if (objToGround) {
+                if (std::find(groundedObjs.begin(), groundedObjs.end(), objToGround)
+                    == groundedObjs.end()) {
```

Comment:
Do you mean 
```
if (!std::ranges::any_of(groundedObjs, [&objToGround](const auto& obj) {
    return obj == objToGround;
})) {
```
of cpp 2020?

Reply:
As you note, you cannot yet use `std::ranges::any_of`, but you could use `std::any_of` and friends: https://en.cppreference.com/w/cpp/algorithm/all_any_none_of -- that said, I don't think that's the right answer here. Your code is a perfectly idiomatic way of determining if a value is in a vector. But you're doing it repeatedly within a loop, and std::find on a vector is O(n). Taking a step back, the whole outer loop looks like it's a `std::copy_if`, isn't it? And it seems like you are treating `groundedObjs` like a set, allowing only one copy of each `DocumentObject`. So I'd strongly consider either using a `std::set` from the beginning, or allowing duplicates to get inserted and then round-tripping through a set to remove the duplicates as a post-processing step.

Reply:
Yes right this function (and many others in assemblyObject) should be manipulating `unordered_set` rather than vectors I guess.
I am modifying this one in this PR but leaving the rest since that would increase a lot this PR.